### ISSUE TITLE
Way to tell if a linop has cumsum operators in it

### ISSUE THREAD

**Issue by asgeirbirkis** [chebfun/chebfun#1392]
At the moment, once we start combining integral and differential operators, there is no way of telling that a linop has a cumsum operator in it. In other words, there is no integration analogue of ```diffOrder```, we use negative ```diffOrder``` to indicate that integration is involved, but that breaks down when we start combining operators. The following demonstrates the issue
```
>> C = operatorBlock.cumsum([-1 1]);
>> C.diffOrder
ans =
    -1
>> D = operatorBlock.diff([-1 1]);
>> D.diffOrder
ans =
     1
>> E = C + D;
>> E.diffOrder
ans =
     1
```
I know this is something @hrothgar was interested in last year, is that still the case?  Should we introduce ```cumsumOrder``` to the ```linBlock``` class?

**Comment by asgeirbirkis** [comment]
I'd even stick a bug label to the following:
```
>> C = operatorBlock.cumsum([-1 1]);
>> I = operatorBlock.eye([-1 1]);
>> E = C+I;
>> E.diffOrder
ans =
     0
```

**Comment by nickhale** [comment]
Is there a reason you want to be able to do this? Obviously `.diffOrder` is required for boundary conditions, but I'm not sure there's a need for a `.cumsumOrder`.

Furthermore, I don't think your example above is a bug at all. `.difforder` has always held the largest differential order in the operator, and here 0 > -1.

Probably the main problem here is that `.diffOrder` is barely documented at all:
```
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %% LINBLOCK CLASS PROPERTIES:
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    properties ( Access = public )

        ...
 
        % Used to track differential order.
        diffOrder = 0;

        ...

    end
```

**Comment by asgeirbirkis** [comment]
This came up when trying to fix #1209. To more carefully treat unknown parameters as opposed to functions, we should categorise any variable that's not acted on by differentiation or integration as parameter. The way we have for that now is to check whether the corresponding ```diffOrder``` is 0. However, when I then try
```
L = chebop(@(u) cumsum(u) + u)
```
we get ```L.diffOrder = 0```, which breaks the test.

**Comment by nickhale** [comment]
I think the way we did this in v4 was by keeping track of 'diagonal' (i.e., multiplication only) operators, which is probably equivalent.

**Comment by asgeirbirkis** [comment]
Yes, that might be a more transparent and robust way. ```isDiag``` could be a property of the ```operatorBlock``` class, I'll investigate this option.

**Comment by asgeirbirkis** [comment]
Closed via #1418
